### ISSUE TITLE
UseSpacing sniff breaks with multiple namespaces/uses in a file

### ISSUE THREAD

**Issue by Majkl578** [slevomat/coding-standard#446]
Doctrine usually has functional tests and its mocks/stubs for tickets in the same file. Ocassionally this could mean multiple namespaces and uses per file.

[This file](https://github.com/doctrine/doctrine2/blob/master/tests/Doctrine/Tests/ORM/Functional/Ticket/DDC2084Test.php#L66-L69) is such example, the sniff breaks (`FAILED TO FIX`) and reports:
```
FOUND 3 ERRORS AFFECTING 2 LINES
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 68 | ERROR | [x] Expected 1 lines after last use statement, found 0. (SlevomatCodingStandard.Namespaces.UseSpacing.IncorrectLinesCountAfterLastUse)
 68 | ERROR | [x] Expected 0 lines between same types of use statement, found -2.
    |       |     (SlevomatCodingStandard.Namespaces.UseSpacing.IncorrectLinesCountBetweenSameTypeOfUse)
 69 | ERROR | [x] Expected 1 lines before first use statement, found 0. (SlevomatCodingStandard.Namespaces.UseSpacing.IncorrectLinesCountBeforeFirstUse)
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 3 MARKED SNIFF VIOLATIONS AUTOMATICALLY
```
(notice `-2`)

Additionally the following code in the middle of file:
```php
use Doctrine\ORM\Annotation as ORM;
use Doctrine\Tests\OrmFunctionalTestCase;
```
is changed into:
```php
use Doctrine\ORM\Annotation as ORM;

use Doctrine\Tests\OrmFunctionalTestCase;


























```
And then still reported as failed to fix.

**Comment by Majkl578** [comment]
Note that similar issue also affects other sniffs, i.e. `SlevomatCodingStandard.Namespaces.ReferenceUsedNamesOnly.ReferenceViaFullyQualifiedName` puts the `use` into wrong namespace.

**Comment by kukulich** [comment]
I'll fix it in version 5.0.0. It's not possible without BC break.

**Comment by Majkl578** [comment]
Ok, np, it only affects the fixer, check works fine IIRC after manually fixing the problem.

**Comment by kukulich** [comment]
Fixed in https://github.com/slevomat/coding-standard/commit/248bd37cab72cdd383600d165b61378ac655e2c2

**Comment by Majkl578** [comment]
Nice!

cc @lcobucci @carusogabriel.
